sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/BaseScenarioTests.cs

```diff
@@ -0,0 +1,98 @@
+
+namespace DnsResolver.Tests.ScenarioTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Reflection;
+    using Microsoft.Azure.Management.DnsResolver;
+    using Microsoft.Azure.Management.DnsResolver.Models;
+    using Microsoft.Azure.Management.Resources;
+    using Microsoft.Azure.Management.Resources.Models;
+    using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+    using DnsResolver.Tests.Extensions;
+    using DnsResolver.Tests.Helpers;
+    using Xunit.Abstractions;
+
+    public class BaseScenarioTests : IDisposable
+    {
+        public BaseScenarioTests(ITestOutputHelper output)
+        {
+            var testName = GetTestName(output);
+
+            this.TestContext = MockContext.Start(this.GetType().Name, testName);
+            this.ResourceHandler = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+            this.ResourceManagementClient = ClientFactory.GetResourcesClient(this.TestContext, this.ResourceHandler);
+            this.DnsResolverManagementClient = ClientFactory.GetDnsResolverClient(this.TestContext, this.ResourceHandler);
+            this.SubscriptionId = TestEnvironmentFactory.GetTestEnvironment().SubscriptionId;
+        }
+
+        protected MockContext TestContext { get; private set; }
+
+        protected RecordedDelegatingHandler ResourceHandler { get; private set; }
+
+        protected ResourceManagementClient ResourceManagementClient { get; private set; }
+
+        protected DnsResolverManagementClient DnsResolverManagementClient { get; private set; }
+
+        protected string SubscriptionId { get; private set; }
+
+        public virtual void Dispose()
+        {
+            TestContext?.Dispose();
+        }
+
+        protected ResourceGroup CreateResourceGroup(string resourceGroupName = null)
+        {
+            return this.ResourceManagementClient.CreateResourceGroup(resourceGroupName: resourceGroupName);
+        }
+
+        protected DnsResolverModel CreateDnsResolver(string location = null, string subscriptionId = null, string resourceGroupName = null, IDictionary<string, string> tags = null) 
+        {
+            location = location ?? Constants.DnsResolverLocation;
+            subscriptionId = subscriptionId ?? this.SubscriptionId;
+            resourceGroupName = resourceGroupName ?? TestDataGenerator.GenerateResourceGroup().Name;
+
+            return this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: TestDataGenerator.GenerateDnsResolverName(),
+                parameters: TestDataGenerator.GenerateDnsResolver(location: location, subscriptionId: subscriptionId, resourceGroupName: resourceGroupName, tags: tags));
+        }
+
+        protected ICollection<DnsResolverModel> CreateDnsResolvers( string resourceGroupName = null, int numDnsResolvers = 3)
```

Comment:
Extra space in front of `string resourceGroupName`.

Reply:
Removed.